ledger/src/shred/shred_code.rs

```diff
@@ -1,18 +1,25 @@
 use {
-    crate::shred::{
-        common::dispatch,
-        legacy, merkle,
-        traits::{Shred, ShredCode as ShredCodeTrait},
-        CodingShredHeader, Error, ShredCommonHeader, ShredType, DATA_SHREDS_PER_FEC_BLOCK,
-        MAX_DATA_SHREDS_PER_SLOT, SIZE_OF_NONCE,
+    crate::{
+        shred::{
+            common::dispatch,
+            legacy, merkle,
+            traits::{Shred, ShredCode as ShredCodeTrait},
+            CodingShredHeader, Error, ShredCommonHeader, ShredType, DATA_SHREDS_PER_FEC_BLOCK,
+            MAX_DATA_SHREDS_PER_SLOT, SIZE_OF_NONCE,
+        },
+        shredder::ERASURE_BATCH_SIZE,
     },
     solana_sdk::{clock::Slot, packet::PACKET_DATA_SIZE, signature::Signature},
     static_assertions::const_assert_eq,
 };
 
-// See ERASURE_BATCH_SIZE.
-const_assert_eq!(MAX_CODE_SHREDS_PER_SLOT, 32_768 * 17);
-pub(crate) const MAX_CODE_SHREDS_PER_SLOT: usize = MAX_DATA_SHREDS_PER_SLOT * 17;
+const MAX_CODE_TO_DATA_SHRED_RATIO: usize = ERASURE_BATCH_SIZE[1] - 1;
+const_assert_eq!(
+    MAX_CODE_SHREDS_PER_SLOT,
+    32_768 * MAX_CODE_TO_DATA_SHRED_RATIO
+);
+pub(crate) const MAX_CODE_SHREDS_PER_SLOT: usize =
+    MAX_DATA_SHREDS_PER_SLOT * MAX_CODE_TO_DATA_SHRED_RATIO;
```

Comment:
imo no need to define `MAX_CODE_TO_DATA_SHRED_RATIO`.
It can just be:
```rust
const_assert_eq!(MAX_CODE_SHREDS_PER_SLOT, 32_768 * 17);
pub(crate) const MAX_CODE_SHREDS_PER_SLOT: usize =
      MAX_DATA_SHREDS_PER_SLOT * (ERASURE_BATCH_SIZE[1] - 1);
```

A partial goal of `const_assert_eq`s is to see what the constants are equal to instead of having to parse the code or jump elsewhere in the code; so using literals there are fine/better.